app/src/androidTest/java/com/example/codingtest1/MainActivityTest.kt

```diff
@@ -31,7 +31,6 @@ internal class MainActivityTest {
         onView(withId(R.id.etName)).perform(typeText("John"), closeSoftKeyboard())
         onView(withId(R.id.etNumber)).perform(typeText("123456789"), closeSoftKeyboard())
         onView(withId(R.id.etDescription)).perform(typeText("Friend"), closeSoftKeyboard())
-
         onView(withId(android.R.id.button1)).perform(click())
 
```

Comment:
**Consider adding a check to verify that the contact name "John" is correctly displayed in the recycler view list cell after it has been added.**
ex:

`onView(withId(R.id.contactsRecyclerView))`
`.perform(scrollToPosition<RecyclerView.ViewHolder>(getCurrentItemCount() - 1))`
     
`onView(allOf(withId(R.id.name), withText("John")))`


**You might need a separate function to get the recycler view item count to obtain the last index.**

`private fun getCurrentItemCount(): Int {`
 `val itemCount = IntArray(1)`
   ` onView(withId(R.id.contactsRecyclerView)).check { view, _ ->`
       `   if (view is RecyclerView) {`
           `       itemCount[0] = view.adapter?.itemCount ?: 0`
  `    }`
 ` }`
  `  return itemCount[0]`
`}`

**Add below dependencies in the app _gradle_**

// _RecyclerView support for Espresso_
androidTestImplementation 'androidx.test.espresso:espresso-contrib:3.5.1'

**Make sure you have the necessary imports for Espresso and RecyclerView at the top of your test class.**